tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs

```diff
@@ -244,6 +254,37 @@ private bool ChosenDirectoryInDevDrive(string directoryPath)
         return false;
     }
 
+    private string GetDirectoryNameFromPath(string path)
+    {
+        var lastIndex = path.LastIndexOf('\\');
+        return path.Substring(lastIndex + 1);
+    }
+
+    private bool MoveDirectories(string sourceDirectory, string targetDirectory, List<string> relatedCacheDirectories)
+    {
+        foreach (var relatedCacheDirectory in relatedCacheDirectories)
+        {
+            var relatedCacheDirectoryName = GetDirectoryNameFromPath(relatedCacheDirectory);
+            if (!MoveDirectory(relatedCacheDirectory, targetDirectory + "\\Related Directories\\" + relatedCacheDirectoryName))
+            {
+                return false;
+            }
+        }
+
+        return MoveDirectory(sourceDirectory, targetDirectory);
+    }
+
+    private void SetRelatedEnvironmentVariables(List<string> relatedEnvironmentVariablesToBeSet, List<string> relatedCacheDirectories, string directoryPath)
+    {
+        var index = 0;
+        foreach (var relatedEnvironmentVariableToBeSet in relatedEnvironmentVariablesToBeSet)
+        {
+            var relatedCacheDirectoryName = GetDirectoryNameFromPath(relatedCacheDirectories[index]);

```

Comment:
That is a good question, for the current case- yes. Generally speaking I would assume the same because it would be through the Env variables that the paths to these dirs would be determined.